private/src/gst/gstavbvideosrc.cpp

```diff
@@ -0,0 +1,246 @@
+
+    gst_element_class_set_static_metadata(GST_ELEMENT_CLASS(klass),
+            "AVB-SH Source", "Source/Video/Network", "Retrieves MPEG-TS packets from AVB-SH",
+            "Intel");
+
+    gobject_class->finalize = gst_avbvideosrc_finalize;
+    gobject_class->set_property = gst_avbvideosrc_set_property;
+    gobject_class->get_property = gst_avbvideosrc_get_property;
+
+    gstelement_class->provide_clock = gst_avbvideosrc_provide_clock;
+    gstelement_class->change_state = gst_avbvideosrc_change_state;
+
+    push_src_class->create = GST_DEBUG_FUNCPTR(gst_avbvideosrc_create);
+
+    g_object_class_install_property(gobject_class, PROP_STREAM_NAME,
+            g_param_spec_string("stream-name", "Stream name",
+                "Name of the stream", DEFAULT_STREAM_NAME,
+                (GParamFlags)(G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS)));
+}
+
+static void gst_avbvideosrc_set_property(GObject *object, guint prop_id,
+        const GValue *value, GParamSpec *pspec)
+{
+    GstAvbVideoSrc *avbvideosrc = GST_AVBVIDEOSRC(object);
+    (void)pspec;
+
+    if (prop_id == PROP_STREAM_NAME) {
+        free(avbvideosrc->stream_name);
+        avbvideosrc->stream_name = g_value_dup_string(value);
+        GST_DEBUG_OBJECT(avbvideosrc, "Set stream name to [%s]", avbvideosrc->stream_name);
+    }
+}
+
+static void gst_avbvideosrc_get_property(GObject *object, guint prop_id,
+        GValue *value, GParamSpec *pspec)
+{
+    GstAvbVideoSrc *avbvideosrc = GST_AVBVIDEOSRC(object);
+
+    if (prop_id == PROP_STREAM_NAME) {
+        g_value_set_string(value, avbvideosrc->stream_name);
+    } else {
+        G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+    }
+}
+
+static GstStateChangeReturn
+gst_avbvideosrc_change_state(GstElement *element, GstStateChange transition)
+{
+    GstAvbVideoSrc *avbvideosrc = GST_AVBVIDEOSRC(element);
+    GstStateChangeReturn ret = GST_STATE_CHANGE_SUCCESS;
+
+    if (transition == GST_STATE_CHANGE_NULL_TO_READY) {
+        ias_avbvideobridge_result r;
+
+        avbvideosrc->receiver = ias_avbvideobridge_create_receiver("MpegTs_Receiver",
+                avbvideosrc->stream_name);
+        if (!avbvideosrc->receiver) {
+            GST_ERROR_OBJECT(avbvideosrc, "Could not create avbvideobridge receiver [%s]",
+                    avbvideosrc->stream_name);
+            return GST_STATE_CHANGE_FAILURE;
+        }
+        r = ias_avbvideobridge_register_MpegTS_cb(avbvideosrc->receiver,
+                &mpegts_callback, avbvideosrc);
+        if (r != IAS_AVB_RES_OK) {
+            GST_ERROR_OBJECT(avbvideosrc, "Could not register mpegts callback [%d]", r);
+            ias_avbvideobridge_destroy_receiver(avbvideosrc->receiver);
+            avbvideosrc->receiver = NULL;
+            return GST_STATE_CHANGE_FAILURE;
+        }
+    }
+
+    ret = GST_ELEMENT_CLASS(gst_avbvideosrc_parent_class)->change_state(element, transition);
+    if (ret == GST_STATE_CHANGE_FAILURE) {
```

Comment:
I think the question that I made in the previous patch applies here: is is possible to leak the `receiver` here?

Reply:
Nice question. I'll check if I have some assurances here.

Reply:
I went through code change with Ederson and it looks good to me.